### PR TITLE
SequenceLoss keras compatibility

### PR DESCRIPTION
**Pull request by kazemnejad**
Fixing the incompatibility issue between the SequenceLoss and Keras built-in training loops by adding support for dense targets. For more information please refer to #375

### CODE REVIEW

File: tensorflow_addons/seq2seq/loss.py [hunk 1]
```diff
@@ -168,6 +182,11 @@ def __init__(self,
         self.sum_over_batch = sum_over_batch
         self.softmax_loss_function = softmax_loss_function
 
+        # Delete the reduction attribute to inform Keras that it
+        # should call this class by the __call__(...) method.
+        if 'reduction' in dir(self):
```
Comment (guillaumekln) [inline tensorflow_addons/seq2seq/loss.py]: `hasattr` seems more appropriate here?

Comment (kazemnejad) [reply tensorflow_addons/seq2seq/loss.py]: Ops, I forget that. Thanks for the comment. it's now fixed in the next commit.

**Comment by guillaumekln** [comment]
@qlzh727 Can you take a quick look when you have time?
